Plugin/Framework/Search/Request/RequestCleaner.php

```diff
@@ -87,44 +90,27 @@ public function __construct(
     public function afterClean(Cleaner $cleaner, array $requestData)
     {
         if (!Search::isNostoSorting($requestData) || !Search::hasCategoryFilter($requestData)) {
-            $this->logger->debugWithSource(
-                'Nosto sorting not used or not found from request data',
-                $requestData,
-                $this
-            );
+            $this->debugWithSource('Nosto sorting not used or not found from request data', [], $requestData);
             //remove nosto_personalised in case it's a search page
             Search::cleanUpCmpSort($requestData);
             return $requestData;
         }
-        try {
 
-            if ($this->containsCatalogViewQueries($requestData)) {
-                $this->webHandler->handle($requestData);
-            } elseif ($this->containsGraphQlProductSearchQueries($requestData)) {
-                $this->graphqlHandler->handle($requestData);
-            } else {
-                $this->logger->debugWithSource(
-                    sprintf(
-                        'Could not find %s from ES request data',
-                        self::KEY_BIND_TO_QUERY
-                    ),
-                    $requestData,
-                    $this
-                );
-                return $requestData;
-            }
-        } catch (Exception $e) {
-            $this->logger->debugWithSource(
-                'Failed to apply CMP - see exception log(s) for details',
-                $requestData,
-                $this
-            );
-            $this->logger->exception($e);
-        } finally {
-            return $requestData;
+        if ($this->containsCatalogViewQueries($requestData)) {
+            $this->webHandler->handle($requestData);
+        } elseif ($this->containsGraphQlProductSearchQueries($requestData)) {
+            $this->graphqlHandler->handle($requestData);
+        } else {
+            $this->debugWithSource('Could not find %s from ES request data', [self::KEY_BIND_TO_QUERY]);
```

Comment:
@olsi-qose I think this should be an exception too.

Reply:
I'm afraid it will break the category page. The exception will not be handled in the method calling it https://github.com/magento/magento2/blob/2.4-develop/lib/internal/Magento/Framework/Search/Request/Builder.php#L165